patches/tModLoader/Terraria/DataStructures/ExtraJumpData.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Terraria.DataStructures;
+
+/// <summary>
+/// A structure containing fields used to manage extra jumps
+/// </summary>
+public struct ExtraJumpData
+{
+	/// <summary>
+	/// Whether the extra jump can be used. This field is what should be set by equipment
```

Comment:
Yeah, enabled overrides are good, we use this design with `SetCrit()` and `DisableCrit()` in hit modifiers for example, providing a clear override precedence.

You've explained to me that setting `Enabled = false` doesn't clear `JumpAvailable`, so that when the thing disabling the jump expires, the jumps can continue where they left off. This is good, but should be clear to a modder (from documentation on `JumpState`)

I would also recommend adding a variable to `Player` which can disable all double jumps (and maybe another one for flight), as a convenience so a modder doesn't have to loop over all jumps every frame to disable them.

We could use such a flag to implement the flipper swimming jump lockout for example.

Reply:
decided to add a `ExtraJumpState::Disable()` method which works as a forced disabling of the extra jump (prevents `PerformJump` from executing, does not affect `Enabled`) as well as `Player::blockExtraJumps` for easily disabling all extra jumps (it's checked alongside the flipper check)